service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderService.java

```diff
@@ -77,44 +61,40 @@ public void progressCMOToAction(CaseDetails caseDetails, CaseManagementOrder ord
         return data;
     }
 
-    public CaseManagementOrder buildCMOWithHearingDate(DynamicList list, CaseManagementOrder order) {
-        CaseManagementOrder.CaseManagementOrderBuilder builder = order.toBuilder();
-
-        if (list != null) {
-            HearingDateDynamicElement hearingDateDynamicElement = hearingBookingService.getHearingDynamicElement(list);
+    public OrderAction removeDocumentFromOrderAction(OrderAction orderAction) {
+        return orderAction.toBuilder().document(null).build();
+    }
 
-            builder
-                .action(OrderAction.builder()
-                    .nextHearingId(hearingDateDynamicElement.getId())
-                    .nextHearingDate(hearingDateDynamicElement.getDate())
-                    .type(order.getAction().getType())
-                    .build())
-                .build();
+    public CaseManagementOrder addNextHearingToCMO(DynamicList list, CaseManagementOrder order) {
+        if (list == null) {
+            return order;
         }
 
-        return builder.build();
+        HearingDateDynamicElement hearingDateDynamicElement = getHearingDynamicElement(list);
+
+        return order.toBuilder()
+            .nextHearing(NextHearing.builder()
+                .id(hearingDateDynamicElement.getId())
+                .date(hearingDateDynamicElement.getDate())
+                .build())
+            .build();
     }
 
     public String createNextHearingDateLabel(CaseManagementOrder caseManagementOrder,
                                              List<Element<HearingBooking>> hearingBookings) {
-        String nextHearingLabel = "";
 
-        if (caseManagementOrder != null && caseManagementOrder.getAction() != null) {
-            UUID nextHearingId = caseManagementOrder.getAction().getNextHearingId();
+        if (caseManagementOrder != null && caseManagementOrder.getNextHearing() != null) {
+            UUID nextHearingId = caseManagementOrder.getNextHearing().getId();
 
-            HearingBooking hearingBooking =
-                hearingBookingService.getHearingBookingByUUID(hearingBookings, nextHearingId);
-
-            nextHearingLabel = formatHearingBookingLabel(hearingBooking.getStartDate());
+            return Optional.ofNullable(hearingBookingService.getHearingBookingByUUID(hearingBookings, nextHearingId))
+                .map(booking -> formatHearingBookingLabel(booking.getStartDate())).orElse("");
```

Comment:
[-1]don't need if statement if you decided to go with optional. Can be replaced with functional flow and you will have only one orElse("")

Reply:
😈FULLY FUNCTIONAL 😈